docs: improve minikube instructions in get started section (#2291)
pulumi/docs: content/docs/get-started/kubernetes/deploy-changes.md

You can see we now have an `ip` [stack output]({{< relref "/docs/intro/concepts/stack#outputs" >}}) that we can `curl` to get the output from the service.

> > **Using Minikube:** Note that `minikube` does not support type LoadBalancer. If you are deploying to `minikube`, see following example to run `kubectl port-forward service/YOUR_SERVICE_NAME 8080:80` command to forward the cluster port to the local machine. Then, we can access the service via `curl http://localhost:8080`, which will get the same result as `curl $(pulumi stack output ip)` as in the environment without using `minikube`.

Reviewer: ```suggestion
> **Using Minikube:** Note that Minikube does not support type `LoadBalancer`. If you are deploying to Minikube, see the following example to run the `kubectl port-forward service/YOUR_SERVICE_NAME 8080:80` command to forward the cluster port to the local machine. Then, the service can be accessed via `curl http://localhost:8080`, which will get the same result as `curl $(pulumi stack output ip)` as in the environment without using Minikube.
```

> 

Author: Thanks you for the review. I have updated these suggestions with a newer commit.

> 

Author: These suggestions violated lint rules.

```text
/home/travis/build/pulumi/docs/content/docs/get-started/kubernetes/deploy-changes.md:

Line 64: Fenced code blocks should be surrounded by blank lines.
```

> 

Author: I have moved the updated paragraph to the end of the document.

> 

Reviewer: @zchase, is there an easy way to relax the linting rules here so we can have code blocks inside the blockquote?

> 

Reviewer 2: @justinvp wrapping the code in a disable comment for the linter should disable everything:
```
<!-- markdownlint-disable blanks-around-fences -->
...
<!-- markdownlint-enable blanks-around-fences -->
```

You can see we now have an `ip` [stack output]({{< relref "/docs/intro/concepts/stack#outputs" >}}) that we can `curl` to get the output from the service.

> **Using Minikube:** Note that Minikube does not support type `LoadBalancer`. If you are deploying to Minikube, see the following example to run the `kubectl port-forward service/YOUR_SERVICE_NAME 8080:80` command to forward the cluster port to the local machine. Then, the service can be accessed via `curl http://localhost:8080`, which will get the same result as `curl $(pulumi stack output ip)` as in the environment without using Minikube.
> ```bash
> $ kubectl get service
> NAME             TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)   AGE
> kubernetes       ClusterIP   10.96.0.1       <none>        443/TCP   26h
> nginx-3hq3kux6   ClusterIP   10.96.185.206   <none>        80/TCP    15m
> ```
>
> ```bash
> $ kubectl port-forward service/nginx-3hq3kux6 8080:80
> Forwarding from 127.0.0.1:8080 -> 80
> Forwarding from [::1]:8080 -> 80

> > ```

Reviewer: Sorry, I should have been more clear: The `>` should only be in front of any empty lines and the lines that start with the three backticks and `bash` to start the code fense.

If `>` is in front of the lines _inside_ the code fence and the line of the closing three backticks, it won't render the way we want:

<img width="747" alt="Screen Shot 2020-01-23 at 12 07 54 PM" src="https://user-images.githubusercontent.com/710598/73019709-fde38c80-3dd8-11ea-8ec3-3400065e22c5.png">

The fix is to write it like this:

~~~markdown
> **Using Minikube:** Note that Minikube does not support type `LoadBalancer`. If you are deploying to Minikube, see the following example to run the `kubectl port-forward service/YOUR_SERVICE_NAME 8080:80` command to forward the cluster port to the local machine. Then, the service can be accessed via `curl http://localhost:8080`, which will get the same result as `curl $(pulumi stack output ip)` as in the environment without using Minikube.
>
> ```bash
$ kubectl get service
NAME             TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)   AGE
kubernetes       ClusterIP   10.96.0.1       <none>        443/TCP   26h
nginx-3hq3kux6   ClusterIP   10.96.185.206   <none>        80/TCP    15m
```
>
> ```bash
$ kubectl port-forward service/nginx-3hq3kux6 8080:80
Forwarding from 127.0.0.1:8080 -> 80
Forwarding from [::1]:8080 -> 80
```
~~~

Which renders the way we want.:

<img width="758" alt="Screen Shot 2020-01-23 at 12 11 06 PM" src="https://user-images.githubusercontent.com/710598/73019933-70ed0300-3dd9-11ea-97aa-42b1c4e6388d.png">

Author: Thank you for the further clarification.

Is it okay if we move the updated paragraph to the bottom of the file?

Reviewer: I'd prefer to keep it in a blockquote, but I am fine with it either in the original spot or at the bottom.

So (disabling the lint rules):

~~~markdown
<!-- markdownlint-disable blanks-around-fences -->
> **Using Minikube:** Note that Minikube does not support type `LoadBalancer`. If you are deploying to Minikube, see the following example to run the `kubectl port-forward service/YOUR_SERVICE_NAME 8080:80` command to forward the cluster port to the local machine. Then, the service can be accessed via `curl http://localhost:8080`, which will get the same result as `curl $(pulumi stack output ip)` as in the environment without using Minikube.
>
> ```bash
$ kubectl get service
NAME             TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)   AGE
kubernetes       ClusterIP   10.96.0.1       <none>        443/TCP   26h
nginx-3hq3kux6   ClusterIP   10.96.185.206   <none>        80/TCP    15m
```
>
> ```bash
$ kubectl port-forward service/nginx-3hq3kux6 8080:80
Forwarding from 127.0.0.1:8080 -> 80
Forwarding from [::1]:8080 -> 80
```
<!-- markdownlint-enable blanks-around-fences -->
~~~

Author: Thank you for the feedback. I have updated the pull request.

You can see we now have an `ip` [stack output]({{< relref "/docs/intro/concepts/stack#outputs" >}}) that we can `curl` to get the output from the service.

<!-- markdownlint-disable blanks-around-fences -->
> **Using Minikube:** Note that Minikube does not support type `LoadBalancer`. If you are deploying to Minikube, see the following example to run the `kubectl port-forward service/YOUR_SERVICE_NAME 8080:80` command to forward the cluster port to the local machine. Then, the service can be accessed via `curl http://localhost:8080`, which will get the same result as `curl $(pulumi stack output ip)` as in the environment without using Minikube.
>
> ```bash
> $ kubectl get service
> NAME             TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)   AGE
> kubernetes       ClusterIP   10.96.0.1       <none>        443/TCP   26h
> nginx-3hq3kux6   ClusterIP   10.96.185.206   <none>        80/TCP    15m
> ```
>
> ```bash
> $ kubectl port-forward service/nginx-3hq3kux6 8080:80
> Forwarding from 127.0.0.1:8080 -> 80
> Forwarding from [::1]:8080 -> 80

> > ```

Reviewer: Still need to remove the `>` _inside_ the code blocks. After that, it should be ready to merge! Thanks, again for the improvements here!

~~~suggestion
$ kubectl get service
NAME             TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)   AGE
kubernetes       ClusterIP   10.96.0.1       <none>        443/TCP   26h
nginx-3hq3kux6   ClusterIP   10.96.185.206   <none>        80/TCP    15m
```
>
> ```bash
$ kubectl port-forward service/nginx-3hq3kux6 8080:80
Forwarding from 127.0.0.1:8080 -> 80
Forwarding from [::1]:8080 -> 80
```
~~~

Author: Hmm. My local Visual Studio Code does not properly render under Markdown preview without these block quote in the code block.

I will change according to your feedback. Pull request updated.

Reviewer: I pulled down the changes in your PR and confirmed that Hugo renders it as expected with these changes. Thanks!.

Author: Great! Good to hear. I did not know that the website is using Hugo. 😅
